src/mocks/handlers.ts

```diff
@@ -269,6 +310,128 @@ LOCATION: http://localhost:3000/_api/Web/Lists(guid'5325476d-8a45-4e66-bdd9-d55d
       ctx.set("request-id", "6b5975a0-40d3-0000-1598-09882fca4612")
     );
   }),
+
+  /**
+   * Get all Roles
+   */
+  rest.get("/_api/web/lists/getByTitle\\('Roles')/items", (req, res, ctx) => {
+    let results = structuredClone(testRoles);
+    return res(
+      ctx.status(200),
+      ctx.delay(responsedelay),
+      ctx.json({ value: results })
+    );
+  }),
+
+  /**
+   * Get a specific Role
+   */
+  rest.get(
+    "/_api/web/lists/getByTitle\\('Roles')/items\\(:ItemId)",
+    (req, res, ctx) => {
+      const { ItemId } = req.params;
+      let result = requests.find((element) => element.Id === Number(ItemId));
+      if (result) {
+        return res(
+          ctx.status(200),
+          ctx.delay(responsedelay),
+          ctx.json({ value: result })
+        );
+      } else {
+        return res(
+          ctx.status(404),
+          ctx.delay(responsedelay),
+          ctx.json(notFound)
+        );
+      }
+    }
+  ),
+
+  /**
+   * Delete a Role
+   */
+  rest.post(
+    "/_api/web/lists/getByTitle\\('Roles')/items\\(:ItemId)",
+    async (req, res, ctx) => {
+      const { ItemId } = req.params;
+      let index = testRoles.findIndex(
+        (element) => element.Id === Number(ItemId)
+      );
+      if (index !== -1) {
+        //let body = await req.json();
+        testRoles.splice(index, 1);
+        console.log(testRoles);
+        return res(
+          ctx.status(200),
+          ctx.delay(responsedelay)
+          //ctx.json({ value: requests[index] })
+        );
+      } else {
+        return res(
+          ctx.status(404),
+          ctx.delay(responsedelay),
+          ctx.json(notFound)
+        );
+      }
+    }
+  ),
+
+  /**
+   * Add a user to a Role
+   * Currently that user will ALWYAS be Brenda Wedding
+   * To update this, we'll need to track users and user ID's
+   */
+  rest.post(
+    "/_api/web/lists/getByTitle\\('Roles')/items",
```

Comment:
Can't quite pinpoint it, but something weird is happening with the Add -- it is wiping out seeing the current user's roles -- so kicks me out of "Manage Roles" so can't confirm addition.